app/v2_1/connectors/PersistenceConnector.scala

```diff
@@ -0,0 +1,380 @@
+    movementEORI: Option[EORINumber],
+    movementReferenceNumber: Option[MovementReferenceNumber],
+    page: Option[PageNumber] = Some(PageNumber(1)),
+    count: ItemCount,
+    receivedUntil: Option[OffsetDateTime] = None,
+    localReferenceNumber: Option[LocalReferenceNumber]
+  )(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[PaginationMovementSummary]
+
+  def postMessage(movementId: MovementId, messageType: Option[MessageType], source: Option[Source[ByteString, _]])(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[UpdateMovementResponse]
+
+  def patchMessage(
+    eoriNumber: EORINumber,
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    body: MessageUpdate
+  )(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Unit]
+
+  def updateMessageBody(
+    messageType: MessageType,
+    eoriNumber: EORINumber,
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    source: Source[ByteString, _]
+  )(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Unit]
+
+  def getMessageBody(eoriNumber: EORINumber, movementType: MovementType, movementId: MovementId, messageId: MessageId)(implicit
+    hc: HeaderCarrier,
+    mat: Materializer,
+    ec: ExecutionContext
+  ): Future[Source[ByteString, _]]
+
+}
+
+@Singleton
+class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, val metrics: MetricRegistry)(implicit appConfig: AppConfig)
+    extends PersistenceConnector
+    with HasMetrics
+    with V2BaseConnector
+    with Logging {
+
+  override def postMovement(eori: EORINumber, movementType: MovementType, source: Option[Source[ByteString, _]])(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[MovementResponse] =
+    withMetricsTimerAsync(MetricsKeys.ValidatorBackend.Post) {
+      _ =>
+        val url = appConfig.movementsUrl.withPath(postMovementUrl(eori, movementType))
+
+        val httpClient = httpClientV2
+          .post(url"$url")
+          .setHeader(Constants.APIVersionHeaderKey -> Constants.APIVersionFinalHeaderValue)
+          .withInternalAuthToken
+
+        (source match {
+          case Some(src) => httpClient.setHeader(HeaderNames.CONTENT_TYPE -> MimeTypes.XML).withBody(src)
+          case None      => httpClient
+        }).executeAndDeserialise[MovementResponse]
+    }
+
+  override def getMessage(eori: EORINumber, movementType: MovementType, movementId: MovementId, messageId: MessageId)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[MessageSummary] = {
+    val url = appConfig.movementsUrl.withPath(getMessageUrl(eori, movementType, movementId, messageId))
+    httpClientV2
+      .get(url"$url")
+      .withInternalAuthToken
+      .setHeader(HeaderNames.CONTENT_TYPE -> MimeTypes.XML)
+      .setHeader(Constants.APIVersionHeaderKey -> Constants.APIVersionFinalHeaderValue)
```

Comment:
as we are not using this additional header in transit-movements so this file can be removed. Please use the v2 file for this connector and service.

Reply:
I propose we leave this in for the time being until we can verify we definitely don't require to identify transitional/final in the database, sending the header and not processing it won't cause any issues